rust/lance/src/dataset.rs

```diff
@@ -1239,6 +1255,32 @@ impl Dataset {
         }
     }
 
+    pub async fn count_unindexed_rows(&self, index_name: &str) -> Result<Option<usize>> {
+        let indices = self.load_indices().await?;
+        let index = indices.iter().find(|idx| idx.name.eq(index_name));
```

Comment:
Minor nit: I think I added a `load_index` method recently.  Though maybe it works by uuid and not name.  We could have a `load_index_by_name`.

Reply:
sure! I've added `load_index_by_name` and called it where needed.